### PR TITLE
[BEAM-2212] Fix printing of file patterns in FileBasedSource

### PR DESCRIPTION
**Pull request by dhalperi**
R: @tgroh

### CODE REVIEW

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java [hunk 1]
```diff
@@ -196,27 +196,26 @@ public final long getEstimatedSizeBytes(PipelineOptions options) throws IOExcept
     // This implementation of method getEstimatedSizeBytes is provided to simplify subclasses. Here
     // we perform the size estimation of files and file patterns using the interface provided by
     // FileSystem.
+    checkState(fileOrPatternSpec.isAccessible(),
+        "Size estimation should be done at execution time.");
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java]: Add something about "The File or Pattern Spec was not available" here?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java [hunk 2]
```diff
@@ -245,14 +244,15 @@ public void populateDisplayData(DisplayData.Builder builder) {
     // split a FileBasedSource based on a file pattern to FileBasedSources based on full single
     // files. For files that can be efficiently seeked, we further split FileBasedSources based on
     // those files to FileBasedSources based on sub ranges of single files.
+    checkState(fileOrPatternSpec.isAccessible(),
+        "Bundle splitting should only happen at execution time.");
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java]: Add something about "The File or Pattern Spec was not available" here?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java (hunk 1 above)
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java]: done

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java (hunk 2 above)
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java]: done

**Comment by dhalperi** [comment]
Thanks! Will self-merge once green.

**Pull request closed without merging**